kedro/io/data_catalog.py

```diff
@@ -186,8 +188,8 @@ def __init__(  # pylint: disable=too-many-arguments
         self.layers = layers
         # Keep a record of all patterns in the catalog.
         # {dataset pattern name : dataset pattern body}
-        self._dataset_patterns = dict(dataset_patterns or {})
-        self._load_versions = dict(load_versions or {})
+        self._dataset_patterns = dataset_patterns or {}
```

Comment:
This means that `dataset_patterns` don't get converted to `dict` anymore right? Hope there's not any weird corner case

Reply:
dataset_patterns is already a dict[str, dict[str, Any]], no?

Reply:
Indeed, I meant to say that I don't know how this function is used - if it's internal and we _know_ that the inputs are dictionaries already, then all is fine 👍🏽